src/truncate.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const _truncate = require('lodash.truncate')
+
+const TRUNCATION_TRAVERSER_MAX_DEPTH = 10
+
+function truncate (value, length, depth) {
+  if (typeof length === 'undefined' || length === null || length < 0) {
+    return value
+  }
+
+  depth = depth || 0
+  if (depth > TRUNCATION_TRAVERSER_MAX_DEPTH) {
+    // skip traversing related to truncation beyond certain depth
+    // this is safe becase we're not losing any data, just not truncating
+    return value
+  }
+
+  const type = typeof value
+
+  if (type === 'object' && value === null) {
+    return value
+  }
+
+  if (type === 'object' && Array.isArray(value)) {
+    const result = []
+    if (value.toString !== Array.prototype.toString) {
+      result.toString = value.toString
+    }
+
+    for (let index = 0; index < value.length; index++) {
+      result[index] = truncate(value[index], length, depth + 1)
+    }
+
+    return result
+  }
+
+  if (type === 'object') {
+    const keys = Object.keys(value)
```

Comment:
Do we accept objects as value? If we do, I don't think we should traverse them and treat each individual child as a value. I think taking it's string representation and truncating that is good enough. Same about array and any other non-string value that should be converted to strings.

Reply:
will do on Monday